src/libostree/ostree-repo-pull.c

```diff
@@ -350,6 +351,30 @@ pull_termination_condition (OtPullData          *pull_data)
   return current_idle;
 }
 
+/* Checks if there is enough free space available so that we can hold onto
+ * caches. Returns FALSE if we are low on disk-space so that appropriate
+ * error could be set to handle cache clean-up (if required).
+ */
+static gboolean
+dir_check_for_free_space (OstreeRepo *repo)
+{
+  int dfd;
+  int free_space;
+  struct statfs st_buf;
+
+  dfd = ostree_repo_get_dfd (repo);
+
+  if (fstatfs (dfd, &st_buf) < 0)
+    return FALSE;
+
+  free_space = st_buf.f_bavail * st_buf.f_bsize >> 30; /* Convert to GB */
```

Comment:
Isn't the first multiplication very likely to overflow on large disks? Also, why does this need a *signed* integer?

Reply:
Also, I have been suggested to keep a granularity acceptable in `MB` too, so I am thinking here to follow `systemd` convention like `200MB` or `2G`. This will require some parsing of the parameter in the config file.